### PR TITLE
CLI to copy Admin Console images to private registry

### PR DESCRIPTION
**Pull request by divolgin**
#### What type of PR is this?

type::feature
#### What this PR does / why we need it:

This is required in order to do auto-upgrades when Admin Console images are used with a private registry.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

This CLI is hidden and will not be documented yet.  Usage looks like this:

```
$ ./bin/kots admin-console push-images docker.io ttl.sh/divolgin
Copying minio/minio:RELEASE.2022-01-08T03-11-54Z to ttl.sh/divolgin/minio:RELEASE.2022-01-08T03-11-54Z
Copying blob 0e3d9ac748ab skipped: already exists  
...
Copying config c0092c1ed2 done  
Writing manifest to image destination
Storing signatures
Copying kotsadm/kotsadm-migrations:v1.59.2 to ttl.sh/divolgin/kotsadm-migrations:v1.59.2
...
Copying config 0ed0ea7b7c done  
Writing manifest to image destination
Storing signatures
...
```

#### Does this PR introduce a user-facing change?

```release-note
* Added experimental CLI for copying Admin Console images to a private registry
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: cmd/kots/cli/admin-console-push-images.go [hunk 1]
```diff
@@ -67,8 +94,45 @@ func AdminPushImagesCmd() *cobra.Command {
 	cmd.Flags().String("registry-username", "", "user name to use to authenticate with the registry")
 	cmd.Flags().String("registry-password", "", "password to use to authenticate with the registry")
 
+	cmd.Flags().Bool("credentials-from-secret", false, "set this flag to use credentials storred in a secret by an existing kotsadm instance")
```
Comment (sgalsaleh) [inline cmd/kots/cli/admin-console-push-images.go:97]: why not just default to using this if the user didn't provide any creds? maybe even for the upgrade command?

Comment (sgalsaleh) [inline cmd/kots/cli/admin-console-push-images.go:97]:
```suggestion
	cmd.Flags().Bool("credentials-from-secret", false, "set this flag to use credentials stored in a secret by an existing kotsadm instance")
```

File: cmd/kots/cli/admin-console-push-images.go [hunk 2]
```diff
@@ -67,8 +94,45 @@ func AdminPushImagesCmd() *cobra.Command {
...
+	cmd.Flags().MarkHidden("credentials-from-secret")
```
Comment (sgalsaleh) [inline cmd/kots/cli/admin-console-push-images.go:98]: why is it hidden? this seems like a great feature :). doesn't this just mean "use the same credentials that were supplied in the install/upgrade command"?

File: cmd/kots/cli/admin-console-push-images.go [hunk 3]
```diff
@@ -55,9 +70,19 @@ func AdminPushImagesCmd() *cobra.Command {
...
-			if err != nil {
-				return errors.Wrap(err, "failed to push images")
+			_, err := os.Stat(airgapArchive)
+			if err == nil {
+				err := kotsadm.PushImages(airgapArchive, options)
+				if err != nil {
+					return errors.Wrap(err, "failed to push images")
+				}
+			} else if os.IsNotExist(err) {
+				err := kotsadm.CopyImages(airgapArchive, options, namespace)
```
Comment (sgalsaleh) [inline cmd/kots/cli/admin-console-push-images.go:80]: took me a second to realize this isn't actually gonna be the airgap archive but the source registry endpoint, right?

Comment (divolgin) [reply cmd/kots/cli/admin-console-push-images.go:80]: yeah, docker.io is the only one right now.  that's where out images are hosted

**Comment by divolgin** [comment]
Another ngrok failure:
> Page 'https://cd85606660c8.ngrok.io/' is not available

**Pull request merged**